qiita_ware/ebi.py

```diff
@@ -591,231 +528,57 @@ def generate_submission_xml(self, action):
         actions = ET.SubElement(submission, 'ACTIONS')
 
         study_action = ET.SubElement(actions, 'ACTION')
-        ET.SubElement(study_action, action, {
+        ET.SubElement(study_action, self.action, {
             'schema': 'study',
             'source': basename(self.study_xml_fp)}
         )
 
         sample_action = ET.SubElement(actions, 'ACTION')
-        ET.SubElement(sample_action, action, {
+        ET.SubElement(sample_action, self.action, {
             'schema': 'sample',
             'source': basename(self.sample_xml_fp)}
         )
 
         experiment_action = ET.SubElement(actions, 'ACTION')
-        ET.SubElement(experiment_action, action, {
+        ET.SubElement(experiment_action, self.action, {
             'schema': 'experiment',
             'source': basename(self.experiment_xml_fp)}
         )
 
         run_action = ET.SubElement(actions, 'ACTION')
-        ET.SubElement(run_action, action, {
+        ET.SubElement(run_action, self.action, {
             'schema': 'run', 'source': basename(self.run_xml_fp)}
         )
 
-        if action == 'ADD':
+        if self.action == 'ADD':
             hold_action = ET.SubElement(actions, 'ACTION')
             ET.SubElement(hold_action, 'HOLD', {
-                'HoldUntilDate': str(date.today() + timedelta(365))}
+                'HoldUntilDate': str(date_to_hold + timedelta(365))}
             )
 
-        return submission_set
+        return ET.tostring(submission_set)
 
-    def _write_xml_file(self, xml_gen_fn, attribute_name, fp,
-                        xml_gen_fn_arg=None):
+    def write_xml_file(self, text, attribute_name, fp):
         """Writes an XML file after calling one of the XML generation
         functions
 
         Parameters
         ----------
-        xml_gen_fn : function
-            The function that will be called to generate the XML that will be
-            written
+        text : str
+            The XML text that will be written
         attribute_name : str
             The name of the attribute in which to store the output filepath
         fp : str
             The filepath to which the XML will be written
-        xml_gen_fn_arg : str, optional
-            Defaults to None. If None, no arguments will be passed to
-            xml_gen_fn. Otherwise, this will be passed as the only argument to
-            xml_gen_fn
-
-        Notes
-        -----
-        xml_gen_fn_arg is needed for generating the submission XML
         """
-        if xml_gen_fn_arg is None:
-            xml_element = xml_gen_fn()
-        else:
-            xml_element = xml_gen_fn(xml_gen_fn_arg)
-
-        xml = minidom.parseString(ET.tostring(xml_element))
+        create_dir(self.xml_dir)
+        xml = minidom.parseString(text)
```

Comment:
The reason for this was to get the XML to look pretty -- see [this SO post](http://stackoverflow.com/questions/749796/pretty-printing-xml-in-python). Of course, there are a lot of caveats on that post, so there may be a better way around it. There is the `lxml` package, but it requires another dependency as I recall, since it relies on some libraries that some (most?) systems didn't seem to have by default. [See the installation instructions here](http://lxml.de/installation.html).